### PR TITLE
Initialize ships

### PR DESCRIPTION
**Pull request by ghost**
Code Review

-initialize board and ships
-place ships at a starting location
-ships have the ability to drag, drop, and rotate 90deg
-ship coordinates are placed on the ship as a reference.  Will be removed later.

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -5,6 +5,8 @@
 	<%= stylesheet_link_tag    "application", media: "all" %>
 	<%= javascript_include_tag "application" %>
 	<%= csrf_meta_tags %>
+  <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
+  <script src="http://ajax.googleapis.com/ajax/libs/jqueryui/1.10.3/jquery-ui.min.js"></script>
```
Comment (kenmazaika) [inline app/views/layouts/application.html.erb:8]: I don't think you should need this to include jquery - that's in there by default.  jqueryui I think makes sense to add here though.

File: app/views/ships/index.html.erb [hunk 2]
```diff
@@ -1 +1,62 @@
...
+      <% if row > 0 %>
+        <div class="square textLabel"><%= row %></div>
+      <% else %>
+        <div class="square"></div>
+      <% end %>
+      <% (1..10).each do |col| %>
+        <% if row == 0 %>
+          <div class="square textLabel"><%= col %></div>  
+        <% else %>  
+          <div class="square" dataCoords="<%= col %><%= row %>">
```
Comment (kenmazaika) [inline app/views/ships/index.html.erb]: I might suggest using `data-column="<%= col %>" data-row="<%= row %>"` to make it easier to fetch.

File: app/assets/javascripts/ship.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  snap: '.square',
+  drop: function(event, ui) {
+    var drop_p = $(this).offset();
+    var drag_p = ui.draggable.offset();
+    var left_end = drop_p.left - drag_p.left + 1;
+    var top_end = drop_p.top - drag_p.top + 1;
+    ui.draggable.animate({
+      top: '+=' + top_end,
+      left: '+=' + left_end
+    });
```
Comment (kenmazaika) [inline app/assets/javascripts/ship.js:15]: In upcoming work, I'd suggest trying to do an AJAX request here to move the piece.

**Review by kenmazaika — APPROVED** [review]
Really awesome work, Kay!  Impressive!

**Comment by ghost** [comment]
Thanks Ken!  I have made the suggested changes, but I am seeing one of the tests is failing.  How can I view the details?

**Comment by kenmazaika** [comment]
When clicking in on the "Details" I see the following:

image

I'd expect the same failures to exist on your environment as well.

**Comment by ghost** [comment]
Thank you.  I forgot Amanda had made changes to the tests.

**Comment by ghost** [comment]
All tests are now passing :)

File: app/controllers/ships_controller.rb [hunk 4]
```diff
@@ -1,17 +1,13 @@
 class ShipsController < ApplicationController
 
   def index
-  end
-
-  def create
-    @ship = Ship.create(ship_params)
-    render action: "index"
+    @ships = Ship.where("game_id = ? AND user_id = ?", params[:game_id], :user_id)
```
Comment (kenmazaika) [inline app/controllers/ships_controller.rb]: I don't think the `:user_id` symbol here is doing what you think.  I think you should instead have:

`current_user.id`

Are the @ships actually used in the create action?

**Comment by kenmazaika** [comment]
Made 1 more comment.  Looks good overall though.  Really solid work, Kay.

**Comment by ghost** [comment]
Made the change to current_user.id in ships controller

**Comment by kenmazaika** [comment]
Nice work!

**Pull request merged**